crates/garble-core/src/evaluator.rs

```diff
@@ -79,6 +126,86 @@ impl Evaluator {
         &'a mut self,
         circ: &'a Circuit,
         inputs: &[Mac],
+    ) -> Result<EncryptedGateConsumer<'a, std::slice::Iter<'a, Gate>>, EvaluatorError> {
+        self.alloc_worker(circ.and_count())?.evaluate(circ, inputs)
+    }
+
+    /// Returns a consumer over batched encrypted gates of a circuit.
+    ///
+    /// # Arguments
+    ///
+    /// * `circ` - The circuit to evaluate.
+    /// * `inputs` - The input labels to the circuit.
+    pub fn evaluate_batched<'a>(
+        &'a mut self,
+        circ: &'a Circuit,
+        inputs: &[Mac],
+    ) -> Result<EncryptedGateBatchConsumer<'a, std::slice::Iter<'a, Gate>>, EvaluatorError> {
+        self.evaluate(circ, inputs).map(EncryptedGateBatchConsumer)
+    }
+
+    /// Returns whether evaluator was set up.
+    pub fn is_setup(&self) -> bool {
+        matches!(self.state, State::Setup(_))
+    }
+}
+
+#[derive(Debug)]
+pub(crate) enum State {
+    Initialized,
+    Setup(Setup),
+    Error,
+}
+
+impl State {
+    pub(crate) fn take(&mut self) -> State {
+        std::mem::replace(self, State::Error)
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct Setup {
+    /// The id to be assigned to the next evaluated AND gate.
+    pub current_gid: u128,
```

Comment:
Fields should almost never be `pub`

```suggestion
    current_gid: u128,
```